runtime/themes/gruvbox.toml

```diff
@@ -1,89 +1,127 @@
-"namespace" = "aqua1"
-"punctuation" = "orange1"
-"punctuation.delimiter" = "orange1"
-"operator" = "purple1"
-"special" = "purple0"
-"variable.other.member" = "blue1"
-"variable" = "fg1"
-"variable.builtin" = "orange1"
-"variable.parameter" = "fg2"
-"type" = "yellow1"
-"type.builtin" = "yellow1"
-"constructor" = { fg = "purple1", modifiers = ["bold"] }
-"function" = { fg = "green1", modifiers = ["bold"] }
-"function.macro" = "aqua1"
-"function.builtin" = "yellow1"
-"tag" = "red1"
-"comment" = { fg = "gray1", modifiers = ["italic"]  }
+"annotation" = { fg = "fg1" }
+
+"attribute" = { fg = "aqua1", modifiers = ["italic"] }
+
+"comment" = { fg = "gray", modifiers = ["italic"] }
+
 "constant" = { fg = "purple1" }
-"constant.builtin" = { fg = "purple1", modifiers = ["bold"] }
-"string" = "green1"
-"constant.numeric" = "purple1"
-"constant.character.escape" = { fg = "fg2", modifiers = ["bold"] }
-"label" = "aqua1"
-"module" = "aqua1"
-
-"diff.plus" = "green1"
-"diff.delta" = "orange1"
-"diff.minus" = "red1"
-
-"warning" = "yellow1"
-"error" = "red1"
-"info" = "aqua1"
-"hint" = "blue1"
+"constant.character" = { fg = "aqua1" }
+"constant.character.escape" = { fg = "orange1" }
+"constant.macro" = { fg = "aqua1" }
+"constructor" = { fg = "purple1" }
+
+"definition" = { underline = { color = "aqua1" } }
+
+"diagnostic" = { underline = { color = "orange1", style = "curl" } }
+"diagnostic.deprecated" = { modifiers = ["crossed_out"] }
+"diagnostic.error" = { underline = { color = "red1", style = "curl" } }
+"diagnostic.hint" = { underline = { color = "blue1", style = "curl" } }
+"diagnostic.info" = { underline = { color = "aqua1", style = "curl" } }
+"diagnostic.warning" = { underline = { color = "yellow1", style = "curl" } }
+# "diagnostic.unnecessary" = { modifiers = ["dim"] }  # do not remove this for future resolving
+
+"error" = { fg = "red1" }
+"hint" = { fg = "blue1" }
+"info" = { fg = "aqua1" }
+"warning" = { fg = "yellow1" }
+
+"diff.delta" = { fg = "yellow1" }
+"diff.minus" = { fg = "red1" }
+"diff.plus" = { fg = "green1" }
+
+"function" = { fg = "green1" }
+"function.builtin" = { fg = "yellow1" }
+"function.macro" = { fg = "blue1" }
+
+"keyword" = { fg = "red1" }
+"keyword.control.import" = { fg = "aqua1" }
+
+"label" = { fg = "red1" }
+
+"markup.bold" = { modifiers = ["bold"] }
+"markup.heading" = "aqua1"
+"markup.italic" = { modifiers = ["italic"] }
+"markup.link.text" = "red1"
+"markup.link.url" = { fg = "green1", modifiers = ["underlined"] }
+"markup.raw" = "red1"
+"markup.strikethrough" = { modifiers = ["crossed_out"] }
+
+"module" = { fg = "aqua1" }
+
+"namespace" = { fg = "fg1" }
+
+"operator" = { fg = "purple1" }
+
+"punctuation" = { fg = "orange1" }
+
+"special" = { fg = "purple0" }
+
+"string" = { fg = "green1" }
+"string.regexp" = { fg = "orange1" }
+"string.special" = { fg = "orange1" }
+"string.symbol" = { fg = "yellow1" }
+
+"tag" = { fg = "aqua1" }
+"tag.attribute" = { fg = "purple" }
+"tag.delimiter" = { fg = "foreground" }
```

Comment:
These colors aren't defined in the palette but also these theme keys aren't highlighted by anything, so these lines can be removed

```suggestion
```

Reply:
@the-mikedavis, please check :)